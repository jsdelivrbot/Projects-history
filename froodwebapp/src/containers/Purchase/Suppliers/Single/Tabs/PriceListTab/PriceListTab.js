import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Row, Col } from 'react-flexbox-grid';
import {
  supplierPriceListGetParallelRequest,
  supplierPriceListUpdateRequest,
} from 'redux-base/actions';
import { Table, Spin, Select, Button } from 'antd';
import { table, row, select } from 'styles/common.scss';
import columns from './priceListTabHelpers';

const mapStateToProps = state => ({
  loadingPage: state.supplier.loadingPage,
  priceLists: state.supplier.priceLists,
  priceListData: state.supplier.priceListData,
  needReloadPriceLists: state.supplier.needReloadPriceLists,
});

const mapDispatchToProps = {
  supplierPriceListGetParallelRequest,
  supplierPriceListUpdateRequest,
};

export class PriceListTab extends Component {

  constructor(props) {
    super(props);
    this.state = {
      selectedPriceLists: [],
      selectedPriceListId: null
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.needReloadPriceLists) {
      this.props.supplierPriceListGetParallelRequest({
        params: [
          { id: this.props.supplierId }
        ]
      });
      return;
    }

    // const { selectedPriceListId } = this.state;

    // if (!selectedPriceListId) {
    //   this.setState({
    //     selectedPriceListId: nextProps.priceListData.priceList.id,
    //     selectedPriceLists: nextProps.priceListData.data
    //   });
    // }
    if (nextProps.priceListData) {
      this.setState({
        selectedPriceListId: nextProps.priceListData.priceList.id,
        selectedPriceLists: nextProps.priceListData.data
      });
    }
  }

  handlePriceListSelect = (selectedPriceListId) => {
    this.setState({ selectedPriceListId });
  }

  handleAttach = () => {
    this.props.supplierPriceListUpdateRequest({
      id: this.props.supplierId,
      payload: {
        priceListId: this.state.selectedPriceListId
      }
    });
  }

  render() {
    const {
      loadingPage,
      priceLists
    } = this.props;

    const {
      selectedPriceListId,
      selectedPriceLists
    } = this.state;
    // get only buy price list type.
    const filteredPriceList = priceLists.filter(item => item.typeDescription === 'Buy');

    return (
      <div>
        <Row end="xs" middle="xs" className={ row }>
          <Col lg={ 1 }>
            Price List
          </Col>
          <Col xs md={ 6 } lg={ 4 }>
            <Select
              className={ select }
              onChange={ this.handlePriceListSelect }
              value={ selectedPriceListId }
            >
              { filteredPriceList.map(priceList => (
                <Select.Option
                  key={ priceList.id }
                  value={ priceList.id }
                >
                  { priceList.name }
                </Select.Option>
              ))}
            </Select>
          </Col>
          <Col lg={ 2 }>
            <Button
              type="primary"
              onClick={ this.handleAttach }
            >
              Attach Price List
            </Button>
          </Col>
          <Col md={ 5 } lg={ 5 } />
        </Row>
        <Spin spinning={ loadingPage }>
          <Row className={ row }>
            <Col xs md lg>
              <Table
                className={ table }
                rowKey="sku"
                size="middle"
                columns={ columns }
                dataSource={ selectedPriceLists }
              />
            </Col>
          </Row>
        </Spin>
      </div>
    );
  }
}

PriceListTab.propTypes = {
  // trigger
  loadingPage: PropTypes.bool.isRequired,
  needReloadPriceLists: PropTypes.bool.isRequired,
  // data
  supplierId: PropTypes.string.isRequired,
  priceLists: PropTypes.array.isRequired,
  priceListData: PropTypes.object.isRequired,
  // redux-base
  supplierPriceListGetParallelRequest: PropTypes.func.isRequired,
  supplierPriceListUpdateRequest: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(PriceListTab);
