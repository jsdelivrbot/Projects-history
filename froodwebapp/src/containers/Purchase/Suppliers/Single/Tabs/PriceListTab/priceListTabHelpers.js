export default [{
  title: 'Product name',
  dataIndex: 'product',
}, {
  title: 'SKU',
  dataIndex: 'sku',
}, {
  title: 'Vendor SKU',
  dataIndex: 'vendorSku',
}, {
  title: 'Buy Price',
  dataIndex: 'price',
}];
