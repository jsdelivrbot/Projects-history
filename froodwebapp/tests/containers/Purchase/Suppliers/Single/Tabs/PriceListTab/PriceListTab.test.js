import { PriceListTab } from 'containers/Purchase/Suppliers/Single/Tabs/PriceListTab/PriceListTab';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('PriceListTab', () => {
  const props = {
    loadingPage: false,
    needReloadPriceLists: false,
    supplierId: '1',
    priceLists: [],
    priceListData: {},
    supplierPriceListGetRequest: jest.fn(),
    supplierPriceListUpdateRequest: jest.fn()
  };

  it('renders the PriceListTab component', () => {
    const component = shallow(<PriceListTab { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of PriceListTab', () => {
    const component = shallow(<PriceListTab { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });

  it('calls supplierPriceListUpdateRequest in handleAttach with correct param', () => {
    const component = shallow(<PriceListTab { ...props } />);
    component.setState({ selectedPriceListId: '1' });
    component.instance().handleAttach();

    expect(props.supplierPriceListUpdateRequest.mock.calls.length).toBe(1);

    expect(props.supplierPriceListUpdateRequest.mock.calls[0][0]).toEqual({
      id: props.supplierId,
      payload: {
        priceListId: '1'
      }
    });

    props.supplierPriceListUpdateRequest.mock.calls.length = 0;
  });
});
