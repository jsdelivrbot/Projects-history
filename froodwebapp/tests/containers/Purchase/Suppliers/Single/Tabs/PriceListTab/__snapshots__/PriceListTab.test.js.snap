// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceListTab capturing Snapshot of PriceListTab 1`] = `
<div>
  <Row
    className="row"
    end="xs"
    middle="xs"
  >
    <Col
      lg={1}
    >
      Price List
    </Col>
    <Col
      lg={4}
      md={6}
      xs={true}
    >
      <Select
        choiceTransitionName="zoom"
        className="select"
        onChange={[Function]}
        prefixCls="ant-select"
        showSearch={false}
        transitionName="slide-up"
        value={null}
      />
    </Col>
    <Col
      lg={2}
    >
      <Button
        clicked={false}
        ghost={false}
        loading={false}
        onClick={[Function]}
        prefixCls="ant-btn"
        type="primary"
      >
        Attach Price List
      </Button>
    </Col>
    <Col
      lg={5}
      md={5}
    />
  </Row>
  <Spin
    prefixCls="ant-spin"
    size="default"
    spinning={false}
    wrapperClassName=""
  >
    <Row
      className="row"
    >
      <Col
        lg={true}
        md={true}
        xs={true}
      >
        <Table
          bordered={false}
          className="table"
          columns={
            Array [
              Object {
                "dataIndex": "product",
                "title": "Product name",
              },
              Object {
                "dataIndex": "sku",
                "title": "SKU",
              },
              Object {
                "dataIndex": "vendorSku",
                "title": "Vendor SKU",
              },
              Object {
                "dataIndex": "price",
                "title": "Buy Price",
              },
            ]
          }
          dataSource={Array []}
          indentSize={20}
          loading={false}
          locale={Object {}}
          prefixCls="ant-table"
          rowKey="sku"
          rowSelection={null}
          showHeader={true}
          size="middle"
          useFixedHeader={false}
        />
      </Col>
    </Row>
  </Spin>
</div>
`;
